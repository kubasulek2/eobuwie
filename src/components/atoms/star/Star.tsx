import {VFC} from 'react';
import {joinClassNames} from '../../../utils/joinClassNames';
import {StarProps} from './types';
import styles from './Star.module.scss';

/**
 * Rating star component, read only, displays empty, full or half full star icon
 */
const Star: VFC<StarProps> = ({classes = [], rating = 1, size = 'medium'}) => {
	const iconName =
		rating === 1
			? styles['star']
			: rating === 0.5
			? styles['star_half']
			: styles['star_empty'];

	const sizeClass =
		size === 'small'
			? styles.small
			: size === 'medium'
			? styles.medium
			: styles.large;

	const classlist = joinClassNames(styles.icon, iconName, ...classes); // custom classes will override defaults
	
	return (
		<div
			data-testid="star"
			role="presentation"
			aria-hidden="true"
			className={sizeClass}
		>
			<svg
				xmlns="http://www.w3.org/2000/svg"
				viewBox="0 0 279.62 265.93"
				className={classlist}
			>
				<polygon
					className={styles.one}
					points="0 101.58 106.8 101.58 139.81 0 139.81 203.15 53.4 265.93 86.41 164.35 0 101.58"
				/>
				<polygon
					className={styles.two}
					points="279.62 101.58 172.81 101.58 139.81 0 139.81 203.15 226.22 265.93 193.21 164.35 279.62 101.58"
				/>
			</svg>
		</div>
	);
};

export default Star;
