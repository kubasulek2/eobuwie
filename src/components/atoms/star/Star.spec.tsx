import { render, screen } from '@testing-library/react';
import Star from './Star';

describe('snapshot', () => {
	it('Matches the snapshot', () => {
    expect(render(<Star />)).toMatchSnapshot();
	});
});



it('Renders star', () => {
  render(<Star />);
  expect(screen.getByTestId('star')).toBeInTheDocument();
});

it('Has presentation role', () => {
  render(<Star />);
  screen.getByRole('presentation', {hidden: true});
});

it('Appends custom classes', () => {
  const classes = ['test1', 'test2'];
  render(<Star classes=
    {classes} />);
  const star = screen.getByRole('presentation', {hidden: true});
  expect(star.firstElementChild!.classList).toContain(classes[0]);
  expect(star.firstElementChild!.classList).toContain(classes[1]);
});


