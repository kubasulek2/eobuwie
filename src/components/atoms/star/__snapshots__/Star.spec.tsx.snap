// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot Matches the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        aria-hidden="true"
        data-testid="star"
        role="presentation"
      >
        <svg
          class="material-icons star medium"
          viewBox="0 0 595.28 841.89"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polygon
            class="one"
            points="162.73 472.56 269.54 472.56 302.54 370.98 302.54 574.13 216.14 636.91 249.14 535.33 162.73 472.56"
          />
          <polygon
            class="two"
            points="442.45 472.56 335.64 472.56 302.64 370.98 302.64 574.13 389.04 636.91 356.04 535.33 442.45 472.56"
          />
        </svg>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      aria-hidden="true"
      data-testid="star"
      role="presentation"
    >
      <svg
        class="material-icons star medium"
        viewBox="0 0 595.28 841.89"
        xmlns="http://www.w3.org/2000/svg"
      >
        <polygon
          class="one"
          points="162.73 472.56 269.54 472.56 302.54 370.98 302.54 574.13 216.14 636.91 249.14 535.33 162.73 472.56"
        />
        <polygon
          class="two"
          points="442.45 472.56 335.64 472.56 302.64 370.98 302.64 574.13 389.04 636.91 356.04 535.33 442.45 472.56"
        />
      </svg>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
